Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut

```diff
@@ -1136,36 +1108,58 @@ void function ThreadedAuthAndConnectToServer( string password = "", bool modsCha
 	if ( NSWasAuthSuccessful() )
 	{
 		// disable all RequiredOnClient mods that are not required by the server and are currently enabled
-		foreach ( string modName in NSGetModNames() )
+		foreach ( ModInfo mod in NSGetModsInformation() )
 		{
-			if ( NSIsModRequiredOnClient( modName ) && NSIsModEnabled( modName ) )
+			string modName = mod.name
+			string modVersion = mod.version
+
+			if ( mod.requiredOnClient && mod.enabled )
 			{
 				// find the mod name in the list of server required mods
 				bool found = false
 				foreach ( RequiredModInfo mod in file.lastSelectedServer.requiredMods )
 				{
-					if (mod.name == modName)
+					// this tolerates a version difference for requiredOnClient core mods (only Northstar.Custom for now)
+					if (mod.name == modName && ( IsCoreMod( modName ) || mod.version == modVersion ))
 					{
 						found = true
+						print(format("\"%s\" (v%s) is required and already enabled.", modName, modVersion))
 						break
 					}
 				}
-				// if we didnt find the mod name, disable the mod
+				// if we didn't find the mod name, disable the mod
 				if (!found)
 				{
 					modsChanged = true
 					NSSetModEnabled( modName, false )
+					print(format("Disabled \"%s\" (v%s) since it's not required on server.", modName, modVersion))
 				}
 			}
 		}
 
 		// enable all RequiredOnClient mods that are required by the server and are currently disabled
 		foreach ( RequiredModInfo mod in file.lastSelectedServer.requiredMods )
 		{
-			if ( NSIsModRequiredOnClient( mod.name ) && !NSIsModEnabled( mod.name ))
+			string modName = mod.name
+			string modVersion = mod.version
+
+			// Tolerate core mods (only Northstar.Custom for now) having a different version than server
+			if ( IsCoreMod(modName) )
+			{
+				string coreModVersion = NSGetModVersions( modName )[0]
+				if ( !NSIsModEnabled( modName ) )
+				{
+					modsChanged = true
+					NSSetModEnabled( modName, true )
+					print(format("Enabled \"%s\" (v%s) to join server.", modName, coreModVersion))
+				}
+			}
+
+			else if ( NSIsModRequiredOnClient( mod.name ) && !NSIsModEnabled( mod.name ))
```

Comment:
Actually, if I remember correctly, server only exposes mods that are client-required through `file.lastSelectedServer.requiredMods`, so the `NSIsModRequiredOnClient( mod.name )` check might be unnecessary.

Reply:
Check was indeed unnecessary and was subsequently removed.